src/Arena.js

```diff
@@ -1,554 +1,555 @@
+  poisonDmg,
+  bleedDmg,
+} from "./actions";
+import {
+  diceRoll,
+  sleep,
+  applyPoisonStyle,
+  oppGetHurt,
+  dodgeShake,
+  oppAttackAnimation,
+  closeDropup,
+  openDropup,
+  bleedTickAnimation,
+  displayDmgNumber,
+  animateBuff
+} from "./data/functions";
+import ClickNHold from "react-click-n-hold";
+import Gladiator from "./models/Gladiator";
+import Opponent from "./models/Opponent";
+
+// PNG assets
+import riposteImage from './assets/images/animations/riposte.gif';
+import { OpponentType } from "./data/opponentData";
 
-let stack = 0
 
 export default function Arena() {
-  
   const dispatch = useDispatch();
-  const battleState = useSelector(state => state.BattleState)
-  const myGladiator = useSelector(state => state.ChosenGladiator.chosen)
-  const opponent = useSelector(state => state.ChosenOpponent.chosenOpp)
-  const opponentArray = useSelector(state => state.OpponentReducer)
-  const allSkills = useSelector(state => state.SkillsReducer)
-  let mySkill = useSelector(state => state.SelectedSkill.mySkill)
-  let skillArray = [];
-  let activeBuffs = battleState.activeBuffs;
+  const {
+    battleState,
+    myGladiator,
+    opponent,
+    opponentArray,
+    allSkills,
+    mySkill,
+    activeBuffs,
+  } = useSelector((state) => ({
+    battleState: state.BattleState,
+    myGladiator: state.ChosenGladiator.chosen,
+    opponent: state.ChosenOpponent.chosenOpp,
+    opponentArray: state.OpponentReducer,
+    allSkills: state.SkillsReducer,
+    mySkill: state.SelectedSkill.mySkill,
+    activeBuffs: state.BattleState.activeBuffs,
+  }));
 
   // Setting skills for use in combat. skillArray will contain all usable skills by gladiator at its current level
-  allSkills.map((skill) => {
-    if (skill.gladiator === myGladiator.name && myGladiator.level >= skill.lvlUnlock && !skill.passive){
-      skillArray.push(skill)
-    }
-    return skillArray
-  });
-
-/*---------------------STAT DECLARATIONS-------------------------*/
-    let opponentHP = opponent.hp + battleState.opponentHealth
-    let currentHP = myGladiator.hp + battleState.playerHealth
-    let myHit
-    let oppHit
-    let atkRoll
-    let critRoll
-    let passiveEffect
-    let bleedTick
-    let myAC = myGladiator.ac
-    let critChance = myGladiator.critChance
-    let dmgBonus = myGladiator.dmgBonus
-    let toHitBonus = myGladiator.toHitBonus
-    let oppToHitBonus = opponent.toHitBonus
-    let blockValue = myGladiator.blockValue
-
-//Applies Dex and STR bonuses to hit and damage
-    if (myGladiator.dex > 15){
-      toHitBonus += (myGladiator.dex - 15)
-    }
+  const skillArray = allSkills.filter(
+    (skill) => skill.gladiator === myGladiator.name && myGladiator.level >= skill.lvlUnlock && !skill.passive
+  );
 
-    if (myGladiator.str > 15){
-      dmgBonus += (myGladiator.str - 15)
-    }
+  const [gladiator] = useState(new Gladiator(myGladiator, activeBuffs, dispatch));
+  const [chosenOpponent] = useState(new Opponent(opponent));
```

Comment:
I didn't flesh out all the use cases for opponent but it's pretty much the same thing